### ISSUE TITLE
Numba example overwrites source clip

### ISSUE THREAD

**Issue by adworacz** [Jaded-Encoding-Thaumaturgy/vs-pyplugin#13]
https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin/blob/25aa243d5b2b777e618a4998668edb92d80e4c75/Examples/Numba/Sigma%20Filter/sigma_filter.py#L64-L65

While running this example, it appears that the clip stored in the `src` variable is overwritten. Specifically, using `vspreview` produces identical (filtered) results between both outputs, even though the first output should be untouched.

In other words, the filter appears to be overwriting the source instead of producing a new, standalone clip.

**Comment by Setsugennoao** [comment]
You quite literally can't do that in vapoursynth.

**Comment by adworacz** [comment]
Here's a video of this (apparently impossible) thing happening, this time with the Numpy example.


https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin/assets/561689/e7804149-d936-42bb-b9aa-67dceeb20801


Perhaps it's a bug with vs-preview? It looks like if I feed this script into `mpv` using `vspipe`, and select the different output indexes, it appears to work correctly.

**Comment by jsaowji** [comment]
smaller repro
```python
from __future__ import annotations

from dataclasses import dataclass

from vssource import source
from vstools import set_output, vs
import functools
from vspyplugin import ProcessMode, PyPluginNumpy


@dataclass
class GaussianFilterData:
    sigma_h: float
    sigma_v: float


class GaussianFilter(PyPluginNumpy[GaussianFilterData]):
    @PyPluginNumpy.process(ProcessMode.SingleSrcIPP)
    def _(self, src: GaussianFilter.DT, dst: GaussianFilter.DT, f: vs.VideoFrame, plane: int, n: int) -> None:
        from scipy.ndimage import gaussian_filter
        gaussian_filter(src, self.fd.sigma_v, output=dst)


def gaussian(clip: vs.VideoNode, sigma: float | tuple[float, float] = 0.5) -> vs.VideoNode:
    if isinstance(sigma, tuple):
        sigma_h, sigma_v = sigma
    else:
        sigma_h = sigma_v = sigma

    return GaussianFilter(clip, sigma_h=sigma_h, sigma_v=sigma_v).invoke()


src = vs.core.std.BlankClip().text.Text("HELOWORLDLDLDLDLLDLDLDLDLDLDLD")
vs.core.std.SetVideoCache(src,1)
src = src.std.ShufflePlanes(0, vs.GRAY)

src.text.Text("a").set_output(0)
gaussian(src, 1.5).text.Text("c").set_output(1)

# open in vspreview seek with "c" selected, switch to "a"
# HELLO text is blurred but not "a"
```
It is probably problem that in here https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin/blob/25aa243d5b2b777e618a4998668edb92d80e4c75/vspyplugin/numpy.py#L222 fout can be cached frame and is probably written to using a readPtr from to_host since no get_write_ptr was called frame data is overridden since copy is COW.
